Formula/v8.rb

```diff
@@ -15,98 +16,213 @@ class V8 < Formula
     sha256 "7bcd1bbd66c11305eeea0c36ca472de8a639f511abe0909c8815b1208dbce7b6" => :mavericks
   end
 
-  option "with-readline", "Use readline instead of libedit"
+  # https://bugs.chromium.org/p/chromium/issues/detail?id=620127
+  depends_on :macos => :el_capitan
 
-  # not building on Snow Leopard:
-  # https://github.com/Homebrew/homebrew/issues/21426
-  depends_on :macos => :lion
-
-  # gyp doesn't run under 2.6 or lower
+  # depot_tools/GN require Python 2.7+
   depends_on "python@2" => :build
 
-  depends_on "readline" => :optional
-
   needs :cxx11
 
-  # Update from "DEPS" file in tarball.
-  # Note that we don't require the "test" DEPS because we don't run the tests.
-  resource "gyp" do
-    url "https://chromium.googlesource.com/external/gyp.git",
-        :revision => "4ec6c4e3a94bd04a6da2858163d40b2429b8aad1"
+  resource "depot_tools" do
+    url "https://chromium.googlesource.com/chromium/tools/depot_tools.git"
```

Comment:
This should probably be pinned to a `revision`.

Reply:
Google recommends always using the latest version of depot_tools. In fact, I believe it will actually update itself. I can pin it if we really want to do that, but I'm not sure it actually makes sense in this case.

Reply:
It mostly just makes me a little nervous that we're intentionally introducing a permanent inconsistency into the build. `depot_tools` is updated regularly enough that someone building from source today could end up with a slightly different set of build bugs or behaviours than someone building from source in 3 days time.

I guess if it's going to update itself anyway & we can't stop that then 🤷‍♂️ , but yeah, deliberate inconsistency in the build process sets me on edge a tiny bit.

Reply:
Yeah...that's just how Google does it though...I'm doing my best to massage their build process into something usable in homebrew. They are a bit at odds because the development pace of V8 is much, much faster. 😅

Reply:
Aye. As (IIRC) the last person to kick the `v8` formula through a significant rewrite I can empathise with the pain of handling anything `v8` in this context 🙃. If we're stuck with it we're stuck with it; unless it causes us hassle I guess I could learn to live with it here.

Reply:
@DomT4 is correct that this needs to be pinned to a specific version. We do not want to make up one morning to a broken build in the stable spec just because of some random change in depot tools.

Reply:
I can pin it, if you insist. It won’t actually do anything though, since depot tools will update itself automatically. 🤷‍♂️

Reply:
That makes me wonder whether we should even be packaging v8.

Reply:
You can disable the auto updating by setting the `DEPOT_TOOLS_UPDATE=0` environment variable.
Updating `depot_tools` to the latest version every time we update v8 should hopefully be enough the ensure compatibility.